### PR TITLE
Fix showing Spark-created Iceberg tables

### PR DESCRIPTION
**Pull request by lxynov**
Fixes https://github.com/trinodb/trino/issues/1592

### CODE REVIEW

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java [hunk 1]
```diff
@@ -354,10 +356,11 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
         List<SchemaTableName> tablesList = schemaName.map(Collections::singletonList)
                 .orElseGet(metastore::getAllDatabases)
                 .stream()
-                .flatMap(schema -> metastore.getTablesWithParameter(schema, TABLE_TYPE_PROP, ICEBERG_TABLE_TYPE_VALUE).stream()
-                        .map(table -> new SchemaTableName(schema, table))
-                        .collect(toList())
-                        .stream())
+                .flatMap(schema -> Stream.concat(
+                        metastore.getTablesWithParameter(schema, TABLE_TYPE_PROP, ICEBERG_TABLE_TYPE_VALUE.toLowerCase(Locale.ENGLISH)).stream()
```
Comment (phd3) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java:360]: can we avoid using two HMS calls here using a disjunctive filter? look at documentation in https://github.com/apache/hive/blob/68a759a9925efe5f04d6dfe8f4a9ccc3d7a068e6/standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/IMetaStoreClient.java#L380

[This PR](https://github.com/trinodb/trino/pull/6595) tries to do it, but it's still a WIP so not 100% sure if it works as expected: https://github.com/trinodb/trino/pull/6595/commits/5c98076f5f1cbccac31f92daf19ef3c7ca934da9#diff-83c934e2538dd1e52ea1dab72378428812209855665ae182bdf523d655ad218eR949

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java [hunk 2]
```diff
@@ -354,10 +356,11 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
         List<SchemaTableName> tablesList = schemaName.map(Collections::singletonList)
                 .orElseGet(metastore::getAllDatabases)
                 .stream()
-                .flatMap(schema -> metastore.getTablesWithParameter(schema, TABLE_TYPE_PROP, ICEBERG_TABLE_TYPE_VALUE).stream()
-                        .map(table -> new SchemaTableName(schema, table))
-                        .collect(toList())
-                        .stream())
+                .flatMap(schema -> Stream.concat(
```
Comment (findepi) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java:359]: You gonna have duplicates when using HMS with case insensitive backend. 
Add `.distinct()` to the steam _within_ `flatMap`.

**Comment by GuoPhilipse** [comment]
@findepi @lxynov i saw the PR was posted almost two months ago, do we have plan for merging this PR recently?

**Comment by findepi** [comment]
@GuoPhilipse i don't think the comments have been applied or replied to, so it would be up to PR author to answer the question.

**Comment by GuoPhilipse** [comment]
> @GuoPhilipse i don't think the comments have been applied or replied to, so it would be up to PR author to answer the question.

Thanks @findepi,@lxynov do you have ideas?

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java (hunk 1 above)
Comment (phd3) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java:360]: This usecase seems to be different from the linked PR in the sense that the change here needs to be at a higher level (in `HiveMetastore` interface). The filter can involve any tree of AND and OR operartors, and so the api design change here will likely need more discussion than just making a small change in `ThriftHiveMetastore`. I'm tending towards prioritizing the correctness (over the performance impact of two calls) for now.

**Comment by phd3** [comment]
Created a separate PR as requested by @lxynov.

**Pull request closed without merging**